kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue

```diff
@@ -163,21 +164,25 @@
         deviceFilters.push(useDeviceChannelFilter({ id: props.filterByChannelId }));
       }
 
-      if (
-        props.filterByFacilityId !== null ||
-        props.filterByFacilityCanSignUp !== null ||
-        props.filterByOnMyOwnFacility !== null
-      ) {
+      const pickNotNull = v => v !== null;
+      // Either we build a facility filter or an empty object.
+      // Passing the empty object to useDeviceFacilityFilter is asking "are there ANY facilities?"
+      const facilityFilter = pickBy(
+        {
+          id: props.filterByFacilityId,
+          learner_can_sign_up: props.filterByFacilityCanSignUp,
+          on_my_own_setup: props.filterByOnMyOwnFacility,
+        },
+        pickNotNull,
+      );
+
+      // If we're filtering a particular facility
+      if (Object.keys(facilityFilter).length > 0) {
         apiParams.subset_of_users_device = false;
-        deviceFilters.push(
-          useDeviceFacilityFilter({
-            id: props.filterByFacilityId,
-            learner_can_sign_up: props.filterByFacilityCanSignUp,
-            on_my_own_setup: props.filterByOnMyOwnFacility,
-          }),
-        );
       }
 
+      deviceFilters.push(useDeviceFacilityFilter(facilityFilter));
```

Comment:
I believe this should be:
```suggestion
      if (Object.keys(facilityFilter).length > 0 || props.filterByHasFacilities) {
        deviceFilters.push(useDeviceFacilityFilter(facilityFilter));
      }
```